File: Vector/vector.cpp
#include <iostream>
#include <cmath>

#include "Vector.h"

using namespace std;

Vector::Vector(double x, double y) {
    this->x = x;
    this->y = y;
}

Vector::~Vector() {}

double Vector::getX() const {
    return x;
}

double Vector::getY() const {
    return y;
}

void Vector::setX(double value) {
    this->x = x;
}

void Vector::setY(double value) {
    this->y = y;
}

double Vector::len(Vector& v) const {
    return hypot(v.x, v.y);
}

bool Vector::operator==(const Vector& other) const {
    return x == other.x && y == other.y;
}

bool Vector::operator!=(const Vector& other) const {
    return *this != other;
}

void Vector::operator+=(const Vector& other) {
    x += other.x;
    y += other.y;
}

void Vector::operator-=(const Vector& other) {
    x -= other.x;
    y -= other.y;
}

Vector Vector::operator+(const Vector& other) const {
    Vector sum;

    sum.x = this->x + other.x;
    sum.y = this->y + other.y;

    return sum;
}

Vector Vector::operator-(const Vector& other) const {
    Vector diff;

    diff.x = this->x - other.x;
    diff.y = this->y - other.y;

    return diff;
}

ostream& operator<<(ostream& out, const Vector& other) {
    return out << '(' << other.getX() << ", " << other.getY() << ')';
}

File: Java/src/com/gymfox/bcwc5/vector/Vector.java
public class Vector implements Cloneable {
    private double x;
    private double y;

    public Vector(double x, double y) {
        this.x = x;
        this.y = y;
    }

    public Vector() {
        this(0.0, 0.0);
    }

    public Vector clone() throws CloneNotSupportedException {
        return (Vector) super.clone();
    }

    public double getX() {
        return x;
    }

    public double getY() {
        return y;
    }

    public void setX(double x) {
        this.x = x;
    }

    public void setY(double y) {
        this.y = y;
    }

    public double distance(Vector other) {
        return Math.hypot(other.x, other.y);
    }

    public boolean equals(Vector other) {
        return x == other.x && y == other.y;
    }

    public Vector sum(Vector other) {
        return new Vector(x - other.x, y - other.y);
    }

    public Vector diff(Vector other) {
        return new Vector(x + other.x, y + other.y);
    }

    public String toString() {
        return "(" + x + ", " + y + ")";
    }
}